code/game/objects/items/weapons/tools.dm

```diff
@@ -497,28 +498,72 @@
 				spawn(100)
 					user.disabilities &= ~NEARSIGHTED
 
+/obj/item/weapon/weldingtool/update_icon()
+	..()
+	if(src.welding)
+		src.icon_state = "welder1"
+	else
+		src.icon_state = "welder"
+
 /obj/item/weapon/weldingtool/empty
 	start_fueled = 0
 
 /obj/item/weapon/weldingtool/largetank
 	name = "Industrial Welding Tool"
+	desc = "The cutting edge between portability and tank size."
+	icon_state = "welder_large"
 	max_fuel = 40
 	starting_materials = list(MAT_IRON = 70, MAT_GLASS = 60)
 	origin_tech = Tc_ENGINEERING + "=2"
 
+/obj/item/weapon/weldingtool/largetank/update_icon()
+	..()
+	if(src.welding)
+		src.icon_state = "welder_large1"
+	else
+		src.icon_state = "welder_large"
```

Comment:
`a ? b : c` is a trinary operator - if `a` is true, then the value of the whole thing is `b`, otherwise it's `c`. So `welding ? "1" : ""` means `if welding, then the string "1", otherwise the empty string`, and that gets embedded as part of the parent string.